### ISSUE TITLE
Enter insulin bolus treatments

### ISSUE THREAD

**Issue by poml88** [nightscout/nightguard#300]
Since we can enter carbs on watch and phone, would it be difficult to implement entering insuline units?

**Comment by dhermanns** [comment]
Hi Phil, can be entered but without any effect. At least for Android APS, since it is not allowed to sync this data to the loop.

**Comment by poml88** [comment]
Well, in my case it is just for keeping track of injections. I have a very simple loop, sensor, brain, pen. :-) I guess I could follow the example of the carbs?

**Comment by dhermanns** [comment]
Yes - should be pretty similar.

**Comment by poml88** [comment]
Just slowly discovering everything... In principle NG shows already everything, and you can enter via the iOS NS view. A quick button like the carbs on the watch would be cool, and a Siri catch like "add 5 units insuline" or so. I'll keep that in mind for later.

**Comment by poml88** [comment]
Ok, in the latest TestFlight there is an option to create Correction Bolus events.

**Comment by poml88** [comment]
Do you want a PR for this? I think you saw the TestFlight?

**Comment by dhermanns** [comment]
I'm not sure whether this could be to dangerous, depending on your setup.

**Comment by poml88** [comment]
You mean, somebody enters 6 U and then this persons pump will inject 6 U and the person will run into a low?
Why would it be different from just entering the insulin in Nightscout? Somebody would do it by mistake?
For me it is very handy, because I see at all times how much insulin is still around...

**Comment by dhermanns** [comment]
Yes exactly. When you setup your own nightscout, it is up to you. But it's different when offering an App in the Appstore.

**Comment by poml88** [comment]
You mean all the "this software comes without any warranty" "no medical decisions" and so on will not help?

Anyway, up to you. I do not know (yet) how all that looping works.

It just makes me developing for you a bit harder, because I will have to keep a different version running for myself. But ok, I guess we will manage. :-)

**Comment by dhermanns** [comment]
No - not really. And then you should hava a look at AAPS. It is a game
changer ;-)

Phil ***@***.***> schrieb am So. 23. Juni 2024 um 18:47:

**Comment by poml88** [comment]
I guess sooner or later I will do look at AAPS. My current game changer is my pancreas that is doing the basal rate for me... I try to keep it alive as long as I can... ;-)

**Comment by dhermanns** [comment]
Fingers crossed 🤞🏽

**Comment by poml88** [comment]
Something else: I have updated the versioning variables, so that the Project and Marketing Versions have to be updated only in one place... Makes it a bit easier (without fast lane?). Would that be of interest?

**Comment by dhermanns** [comment]
Sure

**Comment by dhermanns** [comment]
I would leave this feature on your fork till now. If someone would like to have this, vote for this here with a "+1".

**Comment by poml88** [comment]
Ok, you are the boss. :-) But nobody will every find a closed issue...

**Comment by poml88** [comment]
Just for consideration:
In the carbs view: imagine somebody puts in 20g of carbs and does not eat. Or somebody confuses 10 g with 100 g because of the tiny screen… will also lead to a low. 

I guess making a 100% Safe App is very difficult. Maybe also the reason why for example the fsl apps are quite limited. 

Let’s see what f anybody would like this feature too. :-)

**Comment by dhermanns** [comment]
The effect is much slower and you can simply remove the carbs and correct them. If the insulin is in your body - hard to remove 😉
